ACEHAL/fit.py

```diff
@@ -315,23 +315,22 @@ def do_fit(Psi, Y, B, E0s, solver, n_committee=8, basis_normalization=None, pot_
     if n_committee > 0:
         sigma = solver.sigma_
 
-        # ARD solver returns sigma only for selected features, but does not indicate which.
-        # Reconstruct full signa, hoping that coeff == 0 implies feature was removed by ARD
+        # sklearn ARDRegression solver returns sigma only for selected features, but does not explicitly
+        # indicate which ones those are.
         if sigma.shape[0] != len(c_norm):
-            # make sure sizes match
-            assert sigma.shape[0] == sum(np.nonzero(c_norm))
-
-            sigma = np.zeros((len(c_norm), len(c_norm)), dtype=float)
-            for (i, non_zero) in enumerate(np.nonzero(c_norm)):
-                coeff = solver.sigma_[i, i]
-                sigma[non_zero, non_zero] = coeff
+            # only valid for sklearn ARDRegression
+            included_c = solver.lambda_ < solver.lambda_threshold
+            assert sigma.shape[0] == sum(included_c)
 
+            sigma_full = np.zeros((len(c_norm), len(c_norm)), dtype=sigma.dtype)
+            sigma_full[included_c, included_c] = sigma
```

Comment:
I actually think this doesn't quite work? Anyways it's fixed in the other P+R, however only for the ARD sklearn for now